includes/class-wc-stripe-customer.php

```diff
@@ -441,11 +405,59 @@ public function set_default_source( $source_id ) {
 		return false;
 	}
 
+	/**
+	 * Get the customer saved payment methods using their Stripe ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $type Type of Stripe PaymentMethods to return (card, sepa_debit, bacs_debit, etc)
+	 *
+	 * @return array List of saved payment methods for the customer
+	 * @throws WC_Stripe_Exception  An exception if the customer ID is not valid.
+	 */
+	public function get_payment_methods( $type ) {
+		if ( ! $this->get_id() ) {
+			return [];
+		}
+
+		$payment_methods = false;
+		if ( 'card' === $type || 'sepa_debit' === $type ) {
+			$payment_methods = get_transient( "stripe_payment_methods_{$type}_{$this->get_id()}" );
+		}
```

Comment:
`get_payment_methods` seems to be used only for `card` and `sepa_debit`. Do we need this check `'card' === $type || 'sepa_debit' === $type`? If not, it could be removed from line 447 too.

Reply:
Currently the function is only used to query stripe for `credit cards` and `sepa debits` but it can be used to query the other 12 available payment methods as well.

Previously a single call to `sources` would return all `payment methods`, making caching really simple... now we use a cache per payment type, so we need to make sure we delete all transient that can be possibly created, either by keeping a list of valid payment types and adding a `delete_transient()` call for each one in `clear_cache()`, or explicitly caching only the types we want.